sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,395 @@
+
+In v4:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString){ EntityPath = eventHubName }; 
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+try
+{
+    var batchOptions = new BatchOptions(){ PartitionKey = "my-partition-key" };
+    EventDataBatch eventBatch = partitionSender.CreateBatch(batchOptions);
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+
+    await partitionSender.SendAsync(eventBatch);
+}
+catch()
+{
+    // Handle the exception here
+}
+finally
+{
+    await eventHubClient.CloseAsync();
+}
+```
+
+In v5:
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
+{
+    var batchOptions = new CreateBatchOptions() { PartitionKey = "my-partition-key" };
+    using EventDataBatch eventBatch = await producerClient.CreateBatchAsync(batchOptions);
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("First")));
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
+    
+    await producerClient.SendAsync(eventBatch);
+}
+```
+
+### Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for reading events in batches
+
+In v4, events were read by creating a `PartitionReceiver` and invoking `ReceiveAsync(int)` multiple times to read
+events up to a certain number.
+
+In v5, events can be streamed as they come in without having to use a batched read approach.
+
+This code which reads from a partition in v4:
+
+```csharp
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+var connectionStringBuilder = new EventHubsConnectionStringBuilder(connectionString){ EntityPath = eventHubName }; 
+var eventHubClient = EventHubClient.CreateFromConnectionString(connectionStringBuilder.ToString());
+PartitionReceiver partitionReceiver = client.CreateReceiver("my-consumer-group", "my-partition-id", EventPosition.FromStart());
+try
+{
+    // Gets 100 events or until the read timeout elapses.
+    IEnumerable<EventData> eventDatas = await partitionReceiver.ReceiveAsync(100);
+    // Gets the next 50 events or until the read timeout elapses.
+    IEnumerable<EventData> eventDatas = await partitionReceiver.ReceiveAsync(50);
```

Comment:
Most developers consume messages in a while loop, not this way.

Reply:
Yes, you're right but the intention here is demonstrate developer how to migrate there code from v4  to v5. It is not representative of real-world examples; these are just the code-snippets.

Reply:
Yes, you're right but the intention here is to demonstrate developers how to migrate there code from v4 to v5. It is not representative of real-world examples; these are just the code-snippets.